Add procedure for pods not starting (#1492)
Cray-HPE/docs-csm: troubleshooting/kubernetes/Troubleshoot_Kubernetes_Pods_Not_Starting.md

# Troubleshoot Kubernetes Pods Not Starting

Use this procedure to check if Kubernetes pods get scheduled on an NCN, but don't eventually reach the `Running` state.

## Prerequisites

The `kubectl get pod` command returns pods that seem to be stuck in the `Init` or `ContainerCreating` state.

## Identify the node in question

1. Run the `kubectl get pod -o wide` command to identify the node where the pod isn't starting.

    ```bash

>     ncn-w001# kubectl get pod -A -o wide | grep 'Init\|ContainerCreating'

Reviewer: alternatively you could use egrep 'Init|ContainerCreating'

Author: sounds good -- will change

    services  cray-sls-58cfdb7c46-b7dbj   0/2  Init:0/2           0  2d22h   10.39.0.165  ncn-w002  <none>  <none>
    services  gitea-vcs-65c98746b-jk5v7   0/2  ContainerCreating  0  2d3h    10.47.0.104  ncn-w002  <none>  <none>
    ```

    In the above example, `ncn-w002` is the node that may need attention.

## Recovery Steps

Execute the following steps the node in determined in the previous step.

1. Restart the kubelet service.

   ```bash
   ncn-w002# systemctl restart kubelet
   ```

1. Restart the containerd service.

   ```bash
   ncn-w002# systemctl restart containerd
   ```

> 

Reviewer: should there be a systemctl status or is_active check to make sure it restarted?

Author: great comment, will fix

# Troubleshoot Kubernetes Pods Not Starting

> Use this procedure to check if Kubernetes pods get scheduled on an NCN, but don't eventually reach the `Running` state.

Reviewer 2: ```suggestion
Use this procedure to check if Kubernetes pods get scheduled on an NCN, but do not eventually reach the `Running` state.
```

## Prerequisites

The `kubectl get pod` command returns pods that seem to be stuck in the `Init` or `ContainerCreating` state.

## Identify the node in question

1. Run the `kubectl get pod -o wide` command to identify the node where the pod isn't starting.

    ```bash
    ncn-w001# kubectl get pod -A -o wide | egrep 'Init|ContainerCreating'
    services  cray-sls-58cfdb7c46-b7dbj   0/2  Init:0/2           0  2d22h   10.39.0.165  ncn-w002  <none>  <none>
    services  gitea-vcs-65c98746b-jk5v7   0/2  ContainerCreating  0  2d3h    10.47.0.104  ncn-w002  <none>  <none>
    ```

    In the above example, `ncn-w002` is the node that may need attention.

## Recovery Steps

Execute the following steps the node in determined in the previous step.

> 1. Restart the kubelet service.

Reviewer 2: ```suggestion
1. Restart the `kubelet` service.
```

   ```bash
   ncn-w002# systemctl restart kubelet
   ```

   Now ensure kubelet is running:

   ```bash
   ncn-w002# systemctl status kubelet
   ```

1. Restart the containerd service.

   ```bash
   ncn-w002# systemctl restart containerd
   ```

>    Now ensure containerd is running:

Reviewer 2: ```suggestion
1. Ensure that `containerd` is running.
```
